### PR TITLE
Adds cinnamon, both as plant and associated chem

### PR DESCRIPTION
**Pull request by Glamurio**
[Feature] [Hydroponics]

## About the PR 
This PR adds cinnamon sticks as a vegetable (because technically it is). It grows on cinnamon trees, whose bark is harvested for the delicious treat.

image

Cinnamon is also a chemical that enhances flavor, but has a side effect that when consuming too much of it, it will restrain you in place and make you cough up a bunch of cinnamon dust, rapidly depleting the chem. As this used some of the `/datum/particleSystem/blow_cig_smoke`, I reworked that, as it was always hard-coded to be white smoke, but it now allows for people to pass a desired color for the smoke.

Additionally it adds cinnamon to cinnamon donuts, which had no cinnamon.

## Why's this needed? 
This was a requested feature. Also, cinnamon is delicious.

Relevant thread on the forum:
https://forum.ss13.co/showthread.php?tid=21599

I'm likely going to add cinnamon in powdered form, akin to curry powder, to the chef locker, and allow people to produce it as well. But that will be in a follow-up PR.

## Changelog 

```changelog
(u)Ryou (Glamurio) & Zac 
(*)Added cinnamon to the game. Ask your local botanist to grow you some! Careful not too eat too much of it.
```

### CODE REVIEW

**Comment by MintyPhresh** [comment]
thank you so much for not making this a tree mutation

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 1]
```diff
@@ -3698,6 +3698,69 @@ datum
...
+								if (4) message_append = " What an idiot."
+								if (5) message_append = " That's so stupid."
+								if (6) message_append = " Careless."
+								if (7) message_append = " Huh."
+								if (8) message_append = " Why?"
+								if (9) message_append = " Wow!"
+						M.visible_message("<span class='alert'><B>[M]</B> blows smoke right into <B>[target]</B>'s face![message_append]</span>", group = "[M]_blow_smoke_at_[target]")
+
+						var/mob/living/carbon/human/human_target = target
+						if (human_target && rand(1,5) == 1)
```
Comment (TobleroneSwordfish) [inline code/modules/chemistry/Reagents-FoodDrink.dm:3752]:
```suggestion
						if (target)
```
the cast wasn't doing anything and you may as well make them cough 100%

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 2]
```diff
@@ -3698,6 +3698,69 @@ datum
...
+								if (1) message_append = " Ouch!"
+								if (2) message_append = " Looks painful."
+								if (3) message_append = " Whoa!"
+								if (4) message_append = " What an idiot."
+								if (5) message_append = " That's so stupid."
+								if (6) message_append = " Careless."
+								if (7) message_append = " Huh."
+								if (8) message_append = " Why?"
+								if (9) message_append = " Wow!"
+						M.visible_message("<span class='alert'><B>[M]</B> blows smoke right into <B>[target]</B>'s face![message_append]</span>", group = "[M]_blow_smoke_at_[target]")
```
Comment (TobleroneSwordfish) [inline code/modules/chemistry/Reagents-FoodDrink.dm:3749]:
```suggestion
						M.visible_message("<span class='alert'><B>[M]</B> blows cinnamon right into <B>[target]</B>'s face![message_append]</span>", group = "[M]_blow_smoke_at_[target]")
```

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 3]
```diff
@@ -3698,6 +3698,69 @@ datum
...
+			fluid_b = 102
+			transparency = 255
+			overdose = 20
+			taste = list("sweet", "aromatic")
+
+			do_overdose(var/severity, var/mob/M, var/mult = 1)
+				if(!M) M = holder.my_atom
+
+				if (prob(80))
+					M.losebreath += (1 * mult)
```
Comment (TobleroneSwordfish) [inline code/modules/chemistry/Reagents-FoodDrink.dm:3717]:
```suggestion
				if (prob(80))
					M.losebreath += (2 * mult)
```
challenging

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 4]
```diff
@@ -3698,6 +3698,69 @@ datum
...
+					volume -= overdose
+					M.restrain_time = TIME + 30
```
Comment (TobleroneSwordfish) [inline code/modules/chemistry/Reagents-FoodDrink.dm:3719]:
```suggestion
					M.changeStatus("stunned", 3 SECONDS)
```

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

**Pull request merged**